### ISSUE TITLE
Filters only run for visible columns (Again)

### ISSUE THREAD

**Issue by eirenik** [angular-ui/ng-grid#2476]
This is a loss of functionality that is present in the previous version. I don't know why you closed this as a non-bug.  

In the previous version, it's possible to use a filter like "a:this b:that" and it will filter regardless of columns being visible.  This is very useful functionality, as the alternative is to filter the entire data array elsewhere and replace the grid's data every time a change in filtering is needed.  That is inefficient and has other side-effects.

The filter for the non-visible column is being provided by setting, for example:
<pre>
$scope.gridApi.grid.columns[1].filters[0].term
</pre>

It would obviously be nice if there were a better way to set that, but that's a different issue.

**Comment by PaulL1** [comment]
It's not a bug, it's an enhancement request.  It seems to me that someone consciously chose to make the filtering per-column and only for visible columns, so I probably closed it on the basis that it wasn't the intended direction.

I can leave this open as a future enhancement, but like any code base enhancements it will depend on someone wanting to contribute a pull request that includes it.

**Comment by PaulL1** [comment]
Looking at some other things in filtering, I was in the http://ui-grid.info/docs/#/tutorial/103_filtering tutorial today.

The third column here has a filter term set in code.  If I make this column hidden, the filter appears to still be being applied.  For what reason do you think that filters aren't applied to non-visible columns?

**Comment by eirenik** [comment]
Maybe something about the noTerm implementation addressed that?  The hidden column I noticed this problem with is one that had a term, but was initially hidden (with no mechanism provided to ever display it).

If "add noTerm" is a work-around, that's cool.  But it's not obvious that one would need to "hide" the term for a column that's not displayed at all to get that column filter to work.  Perhaps this just got fixed at some point without me re-checking it against new code.  (Since it didn't work, I'd adjusted the application a bit to avoid the issue.)
